### ISSUE TITLE
solo button does not properly update the 'circle' indicator of reader status

### ISSUE THREAD

**Issue by shadoof** [dhowe/ReadersJS#142]
to reproduce
* turn off all but one reader
* click another reader's solo button

This will make the new reader solo with focus but will not change the 'circular' indicator as expected (glowing circle)

And, generally, there are still a number of problems with the way that the clickable 'circular' indicators interact with the solo and focus buttons ...

**Comment by dhowe** [comment]
@shadoof can you detail these other problems below ?

**Comment by cqx931** [comment]
Please check https://github.com/dhowe/ReadersJS/pull/144
I have tidied up the code and fix a few bugs regarding to the s/f buttons
When a reader is turned off, it's s/f buttons should be hidden.

**Comment by dhowe** [comment]
Still problems:
1) s/f buttons are not hidden when reader is disabled
2) disabled reader can get focus (see below)

![screen shot 2017-06-04 at 11 09 23 pm](https://cloud.githubusercontent.com/assets/737638/26762809/f065657e-497a-11e7-9453-650492d3fecf.png)

**Comment by cqx931** [comment]
s/f buttons are hidden by me. Have you cleared the cache?
![image](https://cloud.githubusercontent.com/assets/3056154/26767934/d104da8c-49d6-11e7-845d-e9895cfc6844.png)

**Comment by dhowe** [comment]
Correct, my fault.
Though I still don't think this is interface is quite there.
Lets discuss.

**Comment by shadoof** [comment]
@dhowe @cqx931 See new bug #152 – there are a lot of inter-related problems still and they are difficult to enumerate/reproduce. It might be advisable to make the 'circles' indicators and on/off switches only: no double-click for focus (do this with the F button) – to simplify.

**Comment by dhowe** [comment]
```
@cqx931do we have a spec for what the interface is supposed to do? We need this 

Also we should make sure we are storing state for each variable only once, even when it is represented multiple times, as in the 'F'/circle case. One approach would be to make Display and ReaderUI objects that represent 'the truth' . Then use _only_ this info to update the entire display on any change, so that things must always be in sync: 

```
ReaderUI { // one per reader

  var parent; // the reader itself
  var enabled, speed;
  var displayName;
}

ReaderDisplay { // singleton

  var soloedReader, focusedReader;
  var text, style, theme;

  var onUserAction = function() { // called on any user action

     // update vars above
     // loop over readers and update reader vars 

     // now data is updated, so update all interface elements
     updateDisplay(this);
  }

  var updateDisplay = function(readerDisplay) { // updates ui elements
  }
}
```

**Comment by cqx931** [comment]
Please check a first draft for the interface spec:
https://github.com/dhowe/ReadersJS/issues/159

**Comment by dhowe** [comment]
status @cqx931 ?

**Comment by cqx931** [comment]
The current interface should work as the spec describes. 
Display and ReaderUI objects you mentioned above are not implemented yet, I can open a new ticket for that.

**Comment by dhowe** [comment]
was only a suggestion -- if we've are in sync with the spec, then all good
